### ISSUE TITLE
A problem in the final step of Unet. 

### ISSUE THREAD

**Issue by zhaozr1119** [Hans-Lan/DASC7606-A2#2]
After running 'python train_mnist.py', the terminal reported an error as shown in the figure. 
The shapes of tensor x before final_conv and after final_conv were printed as [3, 16, 224, 224] and [3, 2, 224, 224]. Are they correct? 
If they are correct, how to deal with this bug? I think it's weird that the terminal reports 128 channels. 
<img width="992" alt="error_report_terminal" src="https://github.com/user-attachments/assets/6753ca9b-af55-4993-840e-af3d76fd973c">

**Comment by tk930528** [comment]
@zhaozr1119 I encounter the same issue, have you fix it?

**Comment by zhaozr1119** [comment]
> @zhaozr1119 I encounter the same issue, have you fix it?

Yes. When you train the model, it takes only two steps in encoder and decoder respectively while passing through Unet.
